code/modules/vehicles/car.dm

```diff
@@ -0,0 +1,122 @@
+#define CAN_KIDNAP 1
+
+/obj/vehicle/sealed/car
+	var/traits = NONE //Bitflag for special behavior such as kidnapping
+	var/engine_sound = 'sound/vehicles/carrev.ogg'
+	var/last_enginesound_time
+	var/engine_sound_length = 20 //Set this to the length of the engine sound
+	layer = ABOVE_MOB_LAYER
+	anchored = TRUE
+
+/obj/vehicle/sealed/car/generate_actions()
+	. = ..()
+	initialize_controller_action_type(/datum/action/vehicle/sealed/remove_key, VEHICLE_CONTROL_DRIVE)
+	if(traits & CAN_KIDNAP)
+		initialize_controller_action_type(/datum/action/vehicle/sealed/DumpKidnappedMobs, VEHICLE_CONTROL_DRIVE)
+
+/obj/vehicle/sealed/car/MouseDrop_T(atom/dropping, mob/M)
+	if(!usr.canmove || usr.stat || usr.restrained())
+		return FALSE
+	if(ismob(dropping) && M != dropping)
+		var/mob/D = dropping
+		M.visible_message("<span class='warning'>[M] starts forcing [D] into \the [src]!</span>")
+		mob_try_forced_enter(M, D)

```

Comment:
i do not see a single Adjacent() or related check, hope you know what you're doing.